packages/scheduler/src/Scheduler.js

```diff
@@ -133,76 +125,34 @@ function flushTask(task, currentTime) {
         );
         break;
     }
-  } catch (error) {
-    throw error;
   } finally {
     currentPriorityLevel = previousPriorityLevel;
     currentTask = previousTask;
   }
 
-  // A callback may return a continuation. The continuation should be scheduled
-  // with the same priority and expiration as the just-finished callback.
+  // A callback may return a continuation.
   if (typeof continuationCallback === 'function') {
-    var expirationTime = task.expirationTime;
```

Comment:
This whole branch is left over from before the "return a continuation" API existed. Instead of popping the task from the queue, checking if there's a continuation, and then rescheduling it, we can peek at the task and only remove it once it completes. If there's a continuation, then the task remains in its current position.